staging/src/k8s.io/apiserver/pkg/server/networkproxy/config.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networkproxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/apis/apiserver"
+	"k8s.io/apiserver/pkg/apis/apiserver/install"
+	"k8s.io/apiserver/pkg/apis/apiserver/v1alpha1"
+	"sigs.k8s.io/yaml"
+)
+
+var cfgScheme = runtime.NewScheme()
+
+func init() {
+	install.Install(cfgScheme)
+}
+
+// ReadNetworkProxyConfiguration reads the network proxy configuration at the specified path.
+// It returns the loaded network proxy configuration if the input file aligns with the required syntax.
+// If it does not align with the provided syntax, it returns a default configuration which should function as a no-op.
+// It does this to preserve backward compatibility when there was no network proxy configuration.
+// It returns an error if the file did not exist.
+func ReadNetworkProxyConfiguration(configFilePath string) (*apiserver.ConnectivityServiceConfiguration, error) {
```

Comment:
@liggitt we've historically used internal types as decoding targets, but with the push for external version, any reason to avoid the point to point conversion and use an external?